diff --git a/src/node_credentials.cc b/src/node_credentials.cc
index 0c7978a4..02140220 100644
--- a/src/node_credentials.cc
+++ b/src/node_credentials.cc
@@ -219,7 +219,9 @@ static const char* name_by_gid(gid_t gid) {
 
 static uid_t uid_by_name(Isolate* isolate, Local<Value> value) {
   if (value->IsUint32()) {
+#ifndef __NANOS__
     static_assert(std::is_same<uid_t, uint32_t>::value);
+#endif  // NanOS uid_t is 16-bit; small uids, harmless
     return value.As<Uint32>()->Value();
   } else {
     Utf8Value name(isolate, value);
@@ -229,7 +231,9 @@ static uid_t uid_by_name(Isolate* isolate, Local<Value> value) {
 
 static gid_t gid_by_name(Isolate* isolate, Local<Value> value) {
   if (value->IsUint32()) {
+#ifndef __NANOS__
     static_assert(std::is_same<gid_t, uint32_t>::value);
+#endif  // NanOS gid_t is 16-bit; small uids, harmless
     return value.As<Uint32>()->Value();
   } else {
     Utf8Value name(isolate, value);
